plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -187,22 +201,41 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
             }
         }
 
-        createTemporaryZipFileAsync { zipOutput ->
+        val zipFilePath = createTemporaryZipFileAsync { zipfs ->
+            val isPosix = FileSystems.getDefault().supportedFileAttributeViews().contains("posix")
             filesToIncludeFlow.collect { file ->
-                try {
-                    val relativePath = Path(file.path).relativeTo(projectRoot.toNioPath())
-                    zipOutput.putNextEntry(relativePath.toString(), Path(file.path))
-                } catch (e: NoSuchFileException) {
-                    // Noop: Skip if file was deleted
+
+                if (!file.isDirectory) {
+                    val externalFilePath = Path(file.path)
+                    val relativePath = Path(file.path).relativeTo(projectRootPath)
+                    val zipfsPath = zipfs.getPath("/$relativePath")
+                    runBlocking {
```

Comment:
it's on the background thread, but is it possible to run this without blocking the thread?